### PR TITLE
Invisible event fix

### PR DESCRIPTION
**Pull request by ghost**
Fixed issue [50](https://github.com/Quivr/Android-Week-View/issues/50) where Events that began on one day and ended at midnight would not get a proper EventRect and would thus be invisible.

### CODE REVIEW

File: library/src/main/java/com/alamkanak/weekview/WeekViewEvent.java [hunk 1]
```diff
@@ -203,8 +203,14 @@ public int hashCode() {
...
-        endTime.add(Calendar.MILLISECOND, -1);
-        if (!isSameDay(this.getStartTime(), endTime)) {
+        if(isAtStartOfNewDay(this.getStartTime(), endTime)) {
+            // Set end time to 1 minute before midnight to ensure the EventRect will get drawn correctly
+            endTime.set(Calendar.HOUR_OF_DAY, 23);
+            endTime.set(Calendar.MINUTE, 59);
+            WeekViewEvent event1 = new WeekViewEvent(this.getId(), this.getName(), this.getLocation(), this.getStartTime(), endTime, this.isAllDay());
+            event1.setColor(this.getColor());
+            events.add(event1);
+        }else if (!isSameDay(this.getStartTime(), endTime)) {
```
Comment (jhoobergs) [inline library/src/main/java/com/alamkanak/weekview/WeekViewEvent.java:213]: @Menthuss  doesn't this mean that the `isSameDay` method works incorrect ? If you fix that  method, you don't need the `isAtStartOfNewDay` part

Comment (jhoobergs) [reply library/src/main/java/com/alamkanak/weekview/WeekViewEvent.java:213]: @Menthuss Any thoughts ?

**Comment by ghost** [comment]
@jhoobergs Hi, sorry for the late response. The reason why I made another method is because the `isSameDay()` method is working as intended. The problem lies in the fact that when an event ends at the start of a new day, the WeekViewEvent won't be given a proper EventRect. This is because the library calculates the size and position of an EventRect based on the difference between the start of the day and the start of the event. You can find a detailed explanation [here](https://github.com/Quivr/Android-Week-View/issues/50).

**Comment by jhoobergs** [comment]
@Menthuss I think that the `isSameDay()` method is ment to also include the `isAtStartOfNewDay ()` method.

**Comment by ghost** [comment]
@jhoobergs I initially thought the same but this would present a problem: If a user were to enter a date that ends at 00:00, `isSameDay()` would now return false and attempt to clone the Event so that it would show over multiple days.

**Comment by jhoobergs** [comment]
@Menthuss Yeah, I was wrong, I thought that a positive result of `isAtStartOfNewDay()` had the same result as `isSameDay()`

**Comment by ghost** [comment]
@jhoobergs Soooo, when will you be able to merge this pull request?

**Comment by ghost** [comment]
> @jhoobergs Soooo, when will you be able to merge this pull request?

**Comment by jhoobergs** [comment]
@Menthuss I will do it in the next couple of days.

**Pull request closed without merging**